### DISCUSSION TITLE
plans for integration with DNS providers?

### DISCUSSION THREAD

**Discussion by stfast** [sys4/netbox-plugin-dns#204] [General]
Hi,

are there any plans to integrate this plugin with DNS providers like Cloudflare or PowerDNS via API?

Or should we do this integration alone with Netbox webhooks?

Regards,
Alex

**Comment by peteeckel** [comment]
Hi @stfast, thanks for raising this, though it's not really an issue but a discussion. I think it makes sense to discuss it in a (hopefully) larger audience, and I also think something useful might come out of this.

First of all: I am a firm believer in modularity and in the philosophy that one tool should do exactly one thing (which might me somewhat larger, but in some way "irreducible") really well and leave the rest to other tools.

The thing NetBox DNS should do very well (and I hope it does) is handle DNS **data**, not servers. Which is broad enough in scoope when you look at what it's currently doing, and will still grow substantially with some other planned enhancements from the roadmap.

Adding support for a large scope of DNS server implementations, be it software like BIND, PowerDNS etc. or "Cloud" services such as Azure, Cloudflare etc., would make the plugin and the required work grow exponentially. Which is bad enough, but worse: Most of the enhancements would be useless for most of the users, while increasing the surface for bugs and vulnerabilities and the time required to keep it all current. So the overall effect would be less code quality and development resources available.

On the other hand, nobody will keep anyone interested from writing another plugin that creates an interface to this or that service. I strongly encourage that kind of work and will do all I can to support it. When you want to create a `netbox-dns-cloudflare` plugin that creates an interface between NetBox DNS and Cloudflare - fine, please do so! The more work in that direction is done the better. 

There is, for example, a plugin [`octodns-netbox-dns`](https://github.com/jcollie/octodns-netbox-dns) that creates an interface to OctoDNS, which in turn can connect to a number of DNS providers that might already do what you need - awesome! If you find something missing in NetBox DNS that is in the way of using this (or any other similar project), please get in touch and we'll find a way to improve it, if necessary with the help of the author of `octodns-netbox-dns`.

**Reply by squintfox** [reply]
@stfast @peteeckel Also, there's a fork that's made some more progress with octodns here: https://github.com/olofvndrhr/octodns-netbox-dns. It's also published on PyPi for convenience.  We're using it currently and it's working great with multiple other octodns providers.
